packages/@aws-cdk/aws-apigatewayv2/lib/http/api.ts

```diff
@@ -457,12 +481,31 @@ export class HttpApi extends HttpApiBase {
    */
   public addRoutes(options: AddRoutesOptions): HttpRoute[] {
     const methods = options.methods ?? [HttpMethod.ANY];
-    return methods.map((method) => new HttpRoute(this, `${method}${options.path}`, {
-      httpApi: this,
-      routeKey: HttpRouteKey.with(options.path, method),
-      integration: options.integration,
-      authorizer: options.authorizer,
-      authorizationScopes: options.authorizationScopes,
-    }));
+    return methods.map((method) => {
+      let authorizationScopes = this.authorizationScopes;
+
+      if (Array.isArray(options.authorizationScopes)) {
```

Comment:
Just checking for truthy should be sufficient, no? Typescript compilation will catch if the type is not an array.

```suggestion
      if (options.authorizationScopes) {
```